src/asm_compiler/mod.rs

```diff
@@ -208,22 +208,43 @@ impl ASMPILConverter {
         start: &usize,
         body: &Vec<InstructionBodyElement>,
         name: &str,
-        params: &Vec<InstructionParam>,
+        params: &InstructionParams,
     ) {
         let instruction_flag = format!("instr_{name}");
         self.create_witness_fixed_pair(*start, &instruction_flag);
         // it's part of the lookup!
         //self.pil.push(constrain_zero_one(&col_name));
 
         let mut substitutions = HashMap::new();
-        for p in params {
-            if p.assignment_reg.0.is_none() && p.assignment_reg.1.is_none() {
-                // literal argument
-                let param_col_name = format!("instr_{name}_param_{}", p.name);
-                self.create_witness_fixed_pair(*start, &param_col_name);
-                substitutions.insert(p.name.clone(), param_col_name);
+        let params = match params {
+            InstructionParams::List(params) => {
+                for p in params {
+                    if p.assignment_reg.0.is_none() && p.assignment_reg.1.is_none() {
```

Comment:
Is it correct that this is always false in the new syntax as we are always looking at either an input or an output?

Reply:
I'm actually not sure if it is useful to have "inout" params, but maybe they could be listed twice in that case...